### PR TITLE
Add doc strings for Block cls

### LINKED ISSUE #689: Add doc strings for Block model
Part of the "define your modules' interfaces" initiative.

### PR DESCRIPTION
**Pull request by TimDaub**
Fixes #689

### CODE REVIEW

File: bigchaindb/models.py [hunk 1]
```diff
@@ -118,8 +118,36 @@ def validate(self, bigchain):
 
 
 class Block(object):
+    """A Block bundles up to 1000 Transactions. Nodes vote on its validity.
+
+        Attributes:
```
Comment (sohkai) [inline bigchaindb/models.py]: Just noticed this now, but all the docstrings in models.py are consistently indented oddly. Normally don't they line up with the first `"`?

File: bigchaindb/models.py [hunk 2]
```diff
@@ -182,15 +212,29 @@ def validate(self, bigchain):
...
 
     def sign(self, signing_key):
+        """Creates a signature for the Block and overwrites `self.signature`.
+
+            Args:
+                signing_key (str): A signing key corresponding to
+                    `self.node_pubkey`.
+
+            Returns:
+                :class:`~.Block`
```
Comment (sohkai) [inline bigchaindb/models.py]: Might be nice to specify that it's still the original `Block` object.

File: bigchaindb/models.py [hunk 3]
```diff
@@ -202,6 +246,24 @@ def is_signature_valid(self):
 
     @classmethod
     def from_dict(cls, block_body):
+        """Transforms a Python dictionary to a Block object.
+
+            Note:
+                Throws errors if passed signature or id is incorrect.
```
Comment (sohkai) [inline bigchaindb/models.py]: I think the `Raises:` part below makes this section redundant.

File: bigchaindb/models.py [hunk 4]
```diff
@@ -237,6 +299,14 @@ def id(self):
         return self.to_dict()['id']
 
     def to_dict(self):
+        """Transforms the object to a Python dictionary.
```
Comment (sohkai) [inline bigchaindb/models.py]: ... the **Block** to a...

File: bigchaindb/models.py [hunk 5]
```diff
@@ -237,6 +299,14 @@ def id(self):
...
+
+            Returns:
+                dict: The Block as an alternative serialization format.
```
Comment (sohkai) [inline bigchaindb/models.py]: ... as **a dict**.

File: bigchaindb/models.py [hunk 6]
```diff
@@ -237,6 +299,14 @@ def id(self):
...
+
+            Raises:
+                OperationError: If a Block doesn't contain any transactions.
```
Comment (sohkai) [inline bigchaindb/models.py]: If **the** Block ...

File: bigchaindb/models.py [hunk 7]
```diff
@@ -118,8 +118,36 @@ def validate(self, bigchain):
 
 
 class Block(object):
+    """A Block bundles up to 1000 Transactions. Nodes vote on its validity.
```
Comment (vrde) [inline bigchaindb/models.py]: If we make the number of transactions per block customizable we need to remember changing this number as well. I'd remove the precise number (`1000`) from the docstring.

File: bigchaindb/models.py [hunk 8]
```diff
@@ -118,8 +118,36 @@ def validate(self, bigchain):
...
+    Attributes:
+        transaction (:obj:`list` of :class:`~.Transaction`):
+            Transactions to be included in the Block.
+        node_pubkey (str): The public key of the node creating the
+            Block.
+        timestamp (str): The Unix time a Block was created.
+        voters (:obj:`list` of :obj:`str`): A list of a federation
+            nodes' public keys supposed to vote on the Block.
+        signature (str): A cryptographic signature ensuring the
+            integrity and creatorship of a Block.
```
Comment (vrde) [inline bigchaindb/models.py]: not sure if `creatorship` is the word we wanna use here. I'd say `A cryptographic signature ensuring the integrity and authenticity of a Block.`

File: bigchaindb/models.py [hunk 9]
```diff
@@ -118,8 +118,36 @@ def validate(self, bigchain):
...
+        Args:
+            transaction (:obj:`list` of :class:`~.Transaction`):
+                Transactions to be included in the Block.
+            node_pubkey (str): The public key of the node creating the
+                Block.
+            timestamp (str): The Unix time a Block was created.
+            voters (:obj:`list` of :obj:`str`): A list of a federation
+                nodes' public keys supposed to vote on the Block.
+            signature (str): A cryptographic signature ensuring the
+                integrity and creatorship of a Block.
```
Comment (vrde) [inline bigchaindb/models.py]: Same as before

**Pull request merged**